ecommerce-app/src/pages/index.js

```diff
@@ -0,0 +1,53 @@
+// pages/index.js
+import React, { useEffect, useState } from "react";
+import Link from "next/link";
+import styles from "../styles/index.module.scss";
+
+const Home = () => {
+  const [categories, setCategories] = useState([]);
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    // Fetch categories
+    fetch("/api/categories")
+      .then((response) => response.json())
+      .then((data) => setCategories(data))
+      .catch((error) => console.error("Error fetching categories:", error));
+
+    // Fetch products
+    fetch("/api/products")
```

Comment:
We should create a next api endpoint that returns both categories and products in a single response. In current approach we are sending two separate requests which is not a best practice. Please make sure you are taking performance into account.